### PR TITLE
feat(storage): move state to persistent storage

### PR DESCRIPTION
**Pull request by gurinderu**

### CODE REVIEW

File: .github/workflows/e2e.yml [hunk 1]
```diff
@@ -55,6 +55,7 @@ jobs:
       - snapshot
     uses: fluencelabs/nox/.github/workflows/build.yml@master
     with:
+      ref: feature/net-759
```
Comment (folex) [inline .github/workflows/e2e.yml:58]: i guess this changes nothing, right

File: .github/workflows/e2e.yml [hunk 2]
```diff
@@ -92,6 +93,7 @@ jobs:
 
     uses: fluencelabs/nox/.github/workflows/tests.yml@master
     with:
+      ref: feature/net-759
```
Comment (folex) [inline .github/workflows/e2e.yml:96]: and this

File: src/spell/modules/spell/spell/src/kv/collection.rs [hunk 3]
```diff
@@ -129,9 +130,10 @@ pub fn list_remove_string(key: &str, value: &str) -> UnitValue {
 #[cfg(test)]
 mod tests {
     use marine_rs_sdk::CallParameters;
+    use marine_rs_sdk::ParticleParameters;
     use marine_rs_sdk_test::marine_test;
 
-    use crate::schema::DB_FILE;
+    const DB_FILE: &str = "./tests_artifacts/spell.sqlite";
```
Comment (folex) [inline src/spell/modules/spell/spell/src/kv/collection.rs:136]: keep or revert?

File: src/spell/modules/spell/spell/src/kv/json.rs [hunk 4]
```diff
@@ -31,10 +31,11 @@ pub fn set_json_fields(json: &str) -> UnitValue {
 #[cfg(test)]
 mod tests {
     use marine_rs_sdk::CallParameters;
+    use marine_rs_sdk::ParticleParameters;
     use marine_rs_sdk_test::marine_test;
     use serde_json::json;
 
-    use crate::schema::DB_FILE;
+    const DB_FILE: &str = "./tests_artifacts/spell.sqlite";
```
Comment (folex) [inline src/spell/modules/spell/spell/src/kv/json.rs:38]: ditto

File: src/spell/modules/spell/spell/src/kv/collection.rs (hunk 3 above)
Comment (gurinderu) [reply src/spell/modules/spell/spell/src/kv/collection.rs:136]: keep

File: src/spell/modules/spell/spell/src/kv/json.rs (hunk 4 above)
Comment (gurinderu) [reply src/spell/modules/spell/spell/src/kv/json.rs:38]: ditto

File: src/spell/modules/spell/spell/test.sh [hunk 5]
```diff
@@ -6,11 +6,10 @@ set -o pipefail
 
 cd "$(dirname "$0")"
 
-rm -f /tmp/spell.sqlite
```
Comment (folex) [inline src/spell/modules/spell/spell/test.sh:9]: sure?

File: src/spell/modules/spell/spell/src/error_handling.rs [hunk 6]
```diff
@@ -212,17 +214,20 @@ mod tests {
 
     /// after_each macro copy-pastes this function into every test
     fn after_each() {
-        std::fs::remove_file(DB_FILE).ok();
+ //       std::fs::remove_file(DB_FILE).ok();
```
[the remark this answers is not in the document]
Comment (gurinderu) [reply src/spell/modules/spell/spell/src/error_handling.rs]: fixed

File: src/spell/modules/spell/spell/src/auth/mod.rs [hunk 7]
```diff
@@ -91,18 +114,37 @@ mod tests {
             tetraplets: vec![],
         };
 
-        assert!(is_kv_write_permitted(HOST_KEY, &cp), "`{}` must be accessible", HOST_KEY);
-        assert!(is_kv_write_permitted(HOST_WORKER_KEY, &cp), "`{}` must be accessible", HOST_WORKER_KEY);
-        assert!(is_kv_write_permitted(WORKER_KEY, &cp), "`{}` must be accessible", WORKER_KEY);
-        assert!(is_kv_write_permitted(PRIVATE_KEY, &cp), "`{}` must be accessible", PRIVATE_KEY);
+        assert!(
```
Comment (kmd-fl) [inline src/spell/modules/spell/spell/src/auth/mod.rs:117]: i see we use different formatters. tbh i prefer a more compact approach but whatever i guess

File: .github/workflows/e2e.yml (hunk 1 above)
Comment (gurinderu) [reply .github/workflows/e2e.yml:58]: It uses a modified nox

File: src/spell/modules/spell/spell/src/kv/primitive.rs [hunk 8]
```diff
@@ -319,37 +330,37 @@ mod tests {
         let cp = || host_call_params();
 
         type SPELL = marine_test_env::spell::ModuleInterface;
-        let set_str_ok = |spell: &mut SPELL, key: &str, value: &str|  {
```
Comment (kmd-fl) [inline src/spell/modules/spell/spell/src/kv/primitive.rs:322]: that's rediculous, why idea didn't do it for me, lol

File: src/spell/modules/spell/spell/src/script.rs [hunk 9]
```diff
@@ -77,10 +77,11 @@ pub fn script_cid() -> CIDv1Value {
 #[test_env_helpers::after_each]
 #[cfg(test)]
 mod tests {
+    use marine_rs_sdk::CallParameters;
+    use marine_rs_sdk::ParticleParameters;
     use marine_rs_sdk_test::marine_test;
-    use marine_rs_sdk_test::CallParameters;
 
-    use crate::schema::DB_FILE;
+    const DB_FILE: &str = "./tests_artifacts/spell.sqlite";
```
Comment (kmd-fl) [inline src/spell/modules/spell/spell/src/script.rs:84]: why don't import it? how does it work now?

**Review by kmd-fl — APPROVED** [review]

File: src/spell/modules/spell/spell/src/auth/mod.rs (hunk 7 above)
Comment (gurinderu) [reply src/spell/modules/spell/spell/src/auth/mod.rs:117]: I used a rustfmt

Comment (kmd-fl) [reply src/spell/modules/spell/spell/src/auth/mod.rs:117]: me too :shrug:

File: src/spell/modules/spell/spell/src/script.rs (hunk 9 above)
Comment (gurinderu) [reply src/spell/modules/spell/spell/src/script.rs:84]: [DB_file](crate::schema::DB_FILE;) it is file inside the wasm runtime, DB_FILE he it is a file in the current system

**Pull request merged**